src/java.base/share/classes/sun/launcher/LauncherHelper.java

```diff
@@ -159,43 +153,34 @@ static void showSettings(boolean printToStderr, String optionFlag,
 
         initOutput(printToStderr);
         String[] opts = optionFlag.split(":");
-        String optStr = opts.length > 1
-                ? opts[1].trim()
-                : "all";
+        String optStr = Arrays.stream(opts).skip(1).findFirst().orElse("").trim();
         switch (optStr) {
-            case "vm":
-                printVmSettings(initialHeapSize, maxHeapSize, stackSize);
-                break;
-            case "properties":
-                printProperties();
-                break;
-            case "locale":
-                printLocale(false);
-                break;
-            case "security":
-                var opt = opts.length > 2 ? opts[2].trim() : "all";
-                SecuritySettings.printSecuritySettings(opt, ostream);
-                break;
-            case "system":
-                if (OperatingSystem.isLinux()) {
-                    printSystemMetrics();
-                    break;
-                }
-            default:
-                printVmSettings(initialHeapSize, maxHeapSize, stackSize);
-                printProperties();
-                printLocale(true);
-                SecuritySettings.printSecuritySummarySettings(ostream);
-                if (OperatingSystem.isLinux()) {
-                    printSystemMetrics();
-                }
-                break;
+            case "all" -> printAllSettings(true, initialHeapSize, maxHeapSize, stackSize);
+            case "locale" -> printLocale(true);
+            case "properties" -> printProperties();
+            case "security" -> SecuritySettings.printSecuritySettings(opts, ostream, true);
+            case "system" -> printSystemMetrics();
+            case "vm" -> printVmSettings(initialHeapSize, maxHeapSize, stackSize);
+            case "" -> printAllSettings(false, initialHeapSize, maxHeapSize, stackSize);
```

Comment:
There's mix in the arguments of where the "verbose" option is included.
Can they be consistent?  (For example put the verbose option as the last argument).